### PR TITLE
[WIP] [Caffe2] Removing extraneous USE_MKL

### PR DESCRIPTION
**Pull request by pjh5**

### CODE REVIEW

File: caffe2/python/CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ set(Caffe2_CPU_PYTHON_SRCS
 )
 
 # ---[ MKL binding
-if(USE_MKL AND USE_MKLML)
+if(USE_MKLML)
```
Comment (yinghai) [inline caffe2/python/CMakeLists.txt]: You need to remove the match `endif()` too.

**Review by yinghai — CHANGES_REQUESTED** [review]

File: caffe2/python/CMakeLists.txt (hunk 1 above)
Comment (pjh5) [reply caffe2/python/CMakeLists.txt]: So remove the if-question entirely?

**Comment by pjh5** [comment]
@yinghai does FindMKL.cmake need to be called to find IDEEP and MKLML, regardless of whether the BLAS has been set to MKL or something else?

**Comment by pjh5** [comment]
It seems these changes are not actually relevant to the Windows issue I was debugging, so I will just leave this for now. These may still be desired changes in the future.

**Comment by pjh5** [comment]
Actually the change in FindMKL.cmake is needed in the pip builds, so I'll continue working on this

**Pull request closed without merging**